packages/config-plugins/src/android/Notifications.ts

```diff
@@ -0,0 +1,145 @@
+  const icon = getNotificationIcon(config);
+  if (icon) {
+    await writeNotificationIconImageFilesAsync(icon, projectRoot);
+  } else {
+    await removeNotificationIconImageFilesAsync(projectRoot);
+  }
+}
+
+export async function setNotificationConfigAsync(
+  config: ExpoConfig,
+  manifestDocument: AndroidManifest
+) {
+  const icon = getNotificationIcon(config);
+  const color = getNotificationColor(config);
+  const mainApplication = getMainApplicationOrThrow(manifestDocument);
+  if (icon) {
+    addMetaDataItemToMainApplication(
+      mainApplication,
+      META_DATA_NOTIFICATION_ICON,
+      NOTIFICATION_ICON_RESOURCE,
+      'resource'
+    );
+  } else {
+    removeMetaDataItemFromMainApplication(mainApplication, META_DATA_NOTIFICATION_ICON);
+  }
+  if (color) {
+    addMetaDataItemToMainApplication(
+      mainApplication,
+      META_DATA_NOTIFICATION_ICON_COLOR,
+      NOTIFICATION_ICON_COLOR_RESOURCE,
+      'resource'
+    );
+  } else {
+    removeMetaDataItemFromMainApplication(mainApplication, META_DATA_NOTIFICATION_ICON_COLOR);
+  }
+  return manifestDocument;
+}
+
+export async function setNotificationIconColorAsync(config: ExpoConfig, projectRoot: string) {
+  const color = getNotificationColor(config);
+  const colorsXmlPath = await Colors.getProjectColorsXMLPathAsync(projectRoot);
+  let colorsJson = await readResourcesXMLAsync({ path: colorsXmlPath });
+  if (color) {
+    const colorItemToAdd = buildResourceItem({ name: NOTIFICATION_ICON_COLOR, value: color });
+    colorsJson = Colors.setColorItem(colorItemToAdd, colorsJson);
+  } else {
+    colorsJson = Colors.removeColorItem(NOTIFICATION_ICON_COLOR, colorsJson);
+  }
+  await writeXMLAsync({ path: colorsXmlPath, xml: colorsJson });
+}
+
+async function writeNotificationIconImageFilesAsync(icon: string, projectRoot: string) {
+  Promise.all(
+    Object.values(dpiValues).map(async ({ folderName, scale }) => {
+      const drawableFolderName = folderName.replace('mipmap', 'drawable');
+      const dpiFolderPath = path.resolve(projectRoot, ANDROID_RES_PATH, drawableFolderName);
+      await fs.ensureDir(dpiFolderPath);
+      const iconSizePx = BASELINE_PIXEL_SIZE * scale;
+
+      try {
+        const resizedIcon = (
+          await generateImageAsync(
+            { projectRoot, cacheType: 'android-notification' },
+            {
+              src: icon,
+              width: iconSizePx,
+              height: iconSizePx,
+              resizeMode: 'cover',
+              backgroundColor: 'transparent',
+            }
+          )
+        ).source;
+        await fs.writeFile(path.resolve(dpiFolderPath, NOTIFICATION_ICON + '.png'), resizedIcon);
+      } catch (e) {
+        throw new Error('Encountered an issue resizing Android notification icon: ' + e);
+      }
+    })
+  );
+}
+
+async function removeNotificationIconImageFilesAsync(projectRoot: string) {
+  Promise.all(
```

Comment:
```suggestion
 await Promise.all(
```

probably want to write a test for this